make sdk fails on ubuntu trusty

when running make sdk on the ubuntu14 vagrant vm, the build fails with following error:

/usr/bin/ld: cannot find -llibosquery_additional

Steps to reproduce:

1. clone osquery
2. launch ubuntu14 vm with vagrant up ubuntu14
3. ssh into the vm and clone the osquery repository
4. cd into the osquery directory and run make deps
5. run make sdk

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -188,6 +188,8 @@ endif()
 # make sdk (tests building SDK-based extensions)
 if(DEFINED ENV{SDK})
   set(OSQUERY_BUILD_SDK_ONLY TRUE)
+else()
+  set(OSQUERY_BUILD_SDK_ONLY FALSE)
 endif()
 
 # make packages will set release to true and blacklist development features,
@@ -417,7 +419,9 @@ add_subdirectory(osquery)
 add_subdirectory(tools/tests)
 
 # Include the kernel building targets/macros
-add_subdirectory(kernel)
+if(NOT ${OSQUERY_BUILD_SDK_ONLY})
+  add_subdirectory(kernel)
+endif()
 
 # make docs
 find_package(Doxygen)

--- a/osquery/CMakeLists.txt
+++ b/osquery/CMakeLists.txt
@@ -151,7 +151,7 @@ add_custom_target(devel
 )
 add_dependencies(devel libosquery)
 
-if(NOT OSQUERY_BUILD_SDK_ONLY)
+if(NOT ${OSQUERY_BUILD_SDK_ONLY})
   # Generate the osquery additional tables (the non-util).
   GENERATE_TABLES("${CMAKE_SOURCE_DIR}")
   AMALGAMATE("${CMAKE_SOURCE_DIR}" "additional" AMALGAMATION)
@@ -219,14 +219,16 @@ if(NOT DEFINED ENV{SKIP_TESTS})
   add_test(osquery_tests osquery_tests)
 
   # osquery kernel tests.
-  add_executable(osquery_kernel_tests main/tests.cpp ${OSQUERY_KERNEL_TESTS})
-  TARGET_OSQUERY_LINK_WHOLE(osquery_kernel_tests libosquery)
-  TARGET_OSQUERY_LINK_WHOLE(osquery_kernel_tests libosquery_additional)
-  target_link_libraries(osquery_kernel_tests gtest libosquery_testing)
-  SET_OSQUERY_COMPILE(osquery_kernel_tests "${CXX_COMPILE_FLAGS} -DKERNEL_TEST=1 -DGTEST_HAS_TR1_TUPLE=0")
+  if(NOT ${OSQUERY_BUILD_SDK_ONLY})
+    add_executable(osquery_kernel_tests main/tests.cpp ${OSQUERY_KERNEL_TESTS})
+    TARGET_OSQUERY_LINK_WHOLE(osquery_kernel_tests libosquery)
+    TARGET_OSQUERY_LINK_WHOLE(osquery_kernel_tests libosquery_additional)
+    target_link_libraries(osquery_kernel_tests gtest libosquery_testing)
+    SET_OSQUERY_COMPILE(osquery_kernel_tests "${CXX_COMPILE_FLAGS} -DKERNEL_TEST=1 -DGTEST_HAS_TR1_TUPLE=0")
+  endif()
 
   # osquery benchmarks.
-  if(NOT DEFINED ENV{SKIP_BENCHMARKS})
+  if(NOT DEFINED ENV{SKIP_BENCHMARKS} AND NOT ${OSQUERY_BUILD_SDK_ONLY})
     add_executable(osquery_benchmarks main/benchmarks.cpp ${OSQUERY_BENCHMARKS})
     TARGET_OSQUERY_LINK_WHOLE(osquery_benchmarks libosquery)
     TARGET_OSQUERY_LINK_WHOLE(osquery_benchmarks libosquery_additional)
@@ -250,7 +252,7 @@ if(NOT DEFINED ENV{SKIP_TESTS})
     )
   endif()
 
-  if(NOT OSQUERY_BUILD_SDK_ONLY)
+  if(NOT ${OSQUERY_BUILD_SDK_ONLY})
     # osquery core (additional) set of unit tests built outside of SDK.
     add_executable(osquery_additional_tests main/tests.cpp ${OSQUERY_ADDITIONAL_TESTS})
     TARGET_OSQUERY_LINK_WHOLE(osquery_additional_tests libosquery)

--- a/osquery/database/plugins/rocksdb.cpp
+++ b/osquery/database/plugins/rocksdb.cpp
@@ -329,7 +329,7 @@ Status RocksDBDatabasePlugin::scan(const std::string& domain,
   auto options = rocksdb::ReadOptions();
   options.verify_checksums = false;
   options.fill_cache = false;
-  auto it = getDB()->NewIterator(rocksdb::ReadOptions(), cfh);
+  auto it = getDB()->NewIterator(options, cfh);
   if (it == nullptr) {
     return Status(1, "Could not get iterator for " + domain);
   }

--- a/osquery/tables/CMakeLists.txt
+++ b/osquery/tables/CMakeLists.txt
@@ -54,7 +54,7 @@ else()
     ADD_OSQUERY_LINK_ADDITIONAL("rpm rpmio")
     ADD_OSQUERY_LINK_ADDITIONAL("cryptsetup devmapper")
     ADD_OSQUERY_LINK_ADDITIONAL("libselinux.so")
-  elseif(UBUNTU)
+  elseif(UBUNTU OR DEBIAN)
     # Ubuntu specific tables
     file(GLOB OSQUERY_UBUNTU_TABLES "*/ubuntu/*.cpp")
     ADD_OSQUERY_LIBRARY_ADDITIONAL(osquery_tables_ubuntu

--- a/tools/deployment/make_linux_package.sh
+++ b/tools/deployment/make_linux_package.sh
@@ -160,10 +160,10 @@ function main() {
   if [[ $OS = "ubuntu" || $OS = "debian" ]]; then
     BUILD_DEBUG_PKG=true
     PACKAGE_DEBUG_NAME="$PACKAGE_NAME-dbg"
-    PACKAGE_DEBUG_DEPENDENCIES="osquery (= $PACKAGE_VERSION)"
+    PACKAGE_DEBUG_DEPENDENCIES="osquery (= $PACKAGE_VERSION-$PACKAGE_ITERATION)"
 
     # Debian only needs the non-stripped binaries.
-    BINARY_DEBUG_DIR=$DEBUG_PREFIX/usr/share/lib/debug/usr/bin
+    BINARY_DEBUG_DIR=$DEBUG_PREFIX/usr/lib/debug/usr/bin
     mkdir -p $BINARY_DEBUG_DIR
     cp "$BUILD_DIR/osquery/osqueryi" $BINARY_DEBUG_DIR
     cp "$BUILD_DIR/osquery/osqueryd" $BINARY_DEBUG_DIR
@@ -202,7 +202,7 @@ function main() {
   PACKAGE_DEBUG_DEPENDENCIES=`echo "$PACKAGE_DEBUG_DEPENDENCIES"|tr '-' '_'`
   OUTPUT_DEBUG_PKG_PATH="$BUILD_DIR/$PACKAGE_DEBUG_NAME-$PACKAGE_VERSION.$PACKAGE_TYPE"
   if [[ $BUILD_DEBUG_PKG ]]; then
-    rm $OUTPUT_DEBUG_PKG_PATH
+    rm -f $OUTPUT_DEBUG_PKG_PATH
     CMD="$FPM -s dir -t $PACKAGE_TYPE \
       -n $PACKAGE_DEBUG_NAME -v $PACKAGE_VERSION \
       --iteration $PACKAGE_ITERATION \

--- a/tools/provision.sh
+++ b/tools/provision.sh
@@ -89,8 +89,10 @@ function main() {
 
   cd "$SCRIPT_DIR/../"
 
-  sudo pip install --upgrade pip
-  sudo pip install -r requirements.txt
+  # Pip may have just been installed.
+  PIP=`which pip`
+  sudo $PIP install --upgrade pip
+  sudo $PIP install -r requirements.txt
 
   initialize $OS
 }

--- a/tools/provision/amazon.sh
+++ b/tools/provision/amazon.sh
@@ -58,6 +58,7 @@ function main_amazon() {
   package flex
   package bison
   package file-libs
+  package file-devel
 
   remove_package libunwind-devel
 

--- a/tools/provision/debian.sh
+++ b/tools/provision/debian.sh
@@ -21,18 +21,13 @@ function main_debian() {
   
   package unzip
   package build-essential
-  package libtool
-  package autoconf
-  package pkg-config
-  
-  package bison
   package flex
   package devscripts
   package debhelper
   package python-pip
   package python-dev
-
   package ruby-dev
+  package rubygems
   package gcc
   package doxygen
 
@@ -46,10 +41,14 @@ function main_debian() {
   package libblkid-dev
   package libbz2-dev
   package libreadline-dev
-  package iptables-dev
-  package libaudit-dev
-  package libmagic-dev
-  
+  package libcurl4-openssl-dev
+
+  package libtool
+  package autoconf
+  package pkg-config
+  package bison
+  package clang
+
   if [[ $DISTRO == "wheezy" ]]; then
     install_cmake
     # thrift requires automate 1.13 or later
@@ -61,13 +60,19 @@ function main_debian() {
   fi 
 
   install_boost
-  install_google_benchmark
-
-  package rubygems
-  gem_install fpm
 
   install_gflags
   install_glog
+  install_google_benchmark
+
+  if [[ $DISTRO == "wheezy" ]]; then
+    gem_install fpm -v 1.3.3
+  else
+    set_cc clang
+    set_cxx clang++
+    gem_install fpm
+  fi
+
   install_snappy
   install_rocksdb 
   install_thrift
@@ -75,12 +80,16 @@ function main_debian() {
   install_asio
   install_cppnetlib
   install_sleuthkit
-  
+
   # Need headers and PC macros
   package libgcrypt-dev
   package libdevmapper-dev
+  package libaudit-dev
+  package libmagic-dev
 
-  package libcryptsetup-dev
+  install_libaptpkg
+  install_iptables_dev
+  install_libcryptsetup
 
   if [[ $DISTRO == "wheezy" ]]; then
     # psutil and other things depending on gcc aren't

--- a/tools/provision/lib.sh
+++ b/tools/provision/lib.sh
@@ -143,6 +143,8 @@ function install_rocksdb() {
       elif [[ $FAMILY = "redhat" ]]; then
         CLANG_VERSION=`clang --version | grep version | cut -d" " -f3`
         CLANG_INCLUDE="-I/usr/lib/clang/$CLANG_VERSION/include"
+        CC=clang
+        CXX=clang++
       fi
       pushd $SOURCE
       if [[ $OS = "freebsd" ]]; then
@@ -152,6 +154,7 @@ function install_rocksdb() {
       else
         MAKE=make
       fi
+      CXXFLAGS="-DROCKSDB_LITE=1 $CXXFLAGS"
       PORTABLE=1 OPT="-DROCKSDB_LITE=1" LIBNAME=librocksdb_lite \
         CC="$CC" CXX="$CXX" CFLAGS="$CLANG_INCLUDE $CFLAGS" CXXFLAGS="$CXXFLAGS" \
         $MAKE -j $THREADS static_lib
@@ -240,9 +243,9 @@ function install_openssl() {
 
   if provision openssl /usr/local/lib/libssl.so.1.0.0; then
     pushd $SOURCE
-    CC="$CC" CXX="$CXX" ./config --prefix=/usr/local --openssldir=/etc/ssl \
-      --libdir=lib shared zlib-dynamic enable-shared \
-      CFLAGS="$CFLAGS" CXXFLAGS="$CXXFLAGS"
+    CFLAGS="$CFLAGS" CXXFLAGS="$CXXFLAGS" CC="$CC" CXX="$CXX" ./config \
+      --prefix=/usr/local --openssldir=/etc/ssl \
+      --libdir=lib shared zlib-dynamic enable-shared
     make -j $THREADS
     sudo make install
     popd
@@ -409,6 +412,7 @@ function install_libcryptsetup() {
     PKG_CONFIG_PATH=/usr/local/lib/pkgconfig/ \
       ./autogen.sh --prefix=/usr/local --enable-static --disable-shared \
         --disable-selinux --disable-udev --disable-veritysetup  --disable-nls \
+        --disable-kernel_crypto \
         CFLAGS="$CFLAGS" CXXFLAGS="$CXXFLAGS"
     pushd lib
     make -j $THREADS

--- a/tools/provision/ubuntu.sh
+++ b/tools/provision/ubuntu.sh
@@ -54,7 +54,6 @@ function main_ubuntu() {
   package libdpkg-dev
   package libudev-dev
   package libblkid-dev
-
   package libbz2-dev
   package libreadline-dev
   package libcurl4-openssl-dev
@@ -118,43 +117,37 @@ function main_ubuntu() {
 
   install_gflags
   install_glog
-  install_iptables_dev
+  install_google_benchmark
 
   if [[ $DISTRO = "lucid" ]]; then
-    gem_install --no-user-install fpm -v 1.3.3
+    gem_install fpm -v 1.3.3
+    install_openssl
+    install_bison
   else
     # No clang++ on lucid
     set_cc clang
     set_cxx clang++
     gem_install fpm
-  fi
-
-  install_snappy
-  install_libaptpkg
-
-  if [[ $DISTRO = "lucid" ]]; then
-    install_openssl
-    install_bison
-  else
     package bison
   fi
 
+  install_snappy
   install_thrift
   install_rocksdb
   install_yara
   install_asio
   install_cppnetlib
-  install_google_benchmark
+  install_sleuthkit
 
   # Need headers and PC macros
   package libgcrypt-dev
   package libdevmapper-dev
-  install_libcryptsetup
-  package libmagic-dev
-  install_sleuthkit
-
-  # Audit facility (kautitd) and netlink APIs
   package libaudit-dev
+  package libmagic-dev
+
+  install_libaptpkg
+  install_iptables_dev
+  install_libcryptsetup
 
   if [[ $DISTRO = "lucid" ]]; then
     package python-argparse

